nft_ingester/src/program_transformers/bubblegum/db.rs

```diff
@@ -114,24 +115,41 @@ where
 pub async fn upsert_asset_with_leaf_info<T>(
     txn: &T,
     id: Vec<u8>,
-    leaf: Option<Vec<u8>>,
-    seq: Option<i64>,
+    nonce: i64,
+    tree_id: Vec<u8>,
```

Comment:
Yep exactly. It also allows us to have a single method `upsert_asset_with_leaf_info` instead of doing different methods when we want to include the tree update. keeps it simple.